scripts/globals/additional_effects.lua

```diff
@@ -173,14 +173,40 @@ xi.additionalEffect.attack = function(attacker, defender, baseAttackDamage, item
         if addStatus and addStatus > 0 then
             local tick   = xi.additionalEffect.statusAttack(addStatus, defender)
             local resist = xi.magic.applyResistanceAddEffect(attacker, defender, element, addStatus, 0)
-            msgID        = xi.msg.basic.ADD_EFFECT_STATUS
+            local immunity = 0
+
+            local immunityTable =
+            {
+                { status = xi.effect.SLEEP_I,   immunity = 0x01   },
+                { status = xi.effect.WEIGHT,    immunity = 0x02   },
+                { status = xi.effect.BIND,      immunity = 0x04   },
+                { status = xi.effect.STUN,      immunity = 0x08   },
+                { status = xi.effect.SILENCE,   immunity = 0x10   },
+                { status = xi.effect.PARALYSIS, immunity = 0x20   },
+                { status = xi.effect.BLINDNESS, immunity = 0x40   },
+                { status = xi.effect.SLOW,      immunity = 0x80   },
+                { status = xi.effect.POISON,    immunity = 0x100  },
+                { status = xi.effect.ELEGY,     immunity = 0x200  },
+                { status = xi.effect.REQUIEM,   immunity = 0x400  },
+                { status = xi.effect.TERROR,    immunity = 0x2000 },
+            }
+
+            for _, statusTable in pairs(immunityTable) do
+                if statusTable.status == addStatus then
+                    immunity = statusTable.immunity
+                    break
+                end
+            end
+
             if
                 resist >= 0.5 and
-                duration * resist > 0
+                duration * resist > 0 and
+                not defender:hasImmunity(immunity)
```

Comment:
hasImmunity takes int values and not hex. You can use xi.immunity.SLEEP, etc. instead of the hex values in the table above